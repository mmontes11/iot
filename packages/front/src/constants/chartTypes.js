export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
export const REALTIME = "REALTIME";
