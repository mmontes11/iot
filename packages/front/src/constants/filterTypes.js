export const THING_FILTER_TYPE = "thing";
export const DATE_FILTER_TYPE = "date";
export const FILTER_TYPES = [THING_FILTER_TYPE, DATE_FILTER_TYPE];
