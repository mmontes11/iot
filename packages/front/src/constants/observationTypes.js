export const EVENT_TYPE = "event";
export const MEASUREMENT_TYPE = "measurement";
export const OBSERVATION_TYPES = [EVENT_TYPE, MEASUREMENT_TYPE];
