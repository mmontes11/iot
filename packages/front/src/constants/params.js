export const TYPE = "TYPE";
export const OBSERVATION = "OBSERVATION";
export const GROUPBY = "GROUPBY";
export const THING = "THING";
