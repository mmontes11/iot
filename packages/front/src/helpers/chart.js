export const colorForIndex = index => {
  const barColors = ["#247ba0", "#70c1b3", "#b2dbbf"];
  return barColors[index % barColors.length];
};
