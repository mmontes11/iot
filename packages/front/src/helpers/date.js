export const formatDateTime = (dateString, formatDate, formatTime) =>
  `${formatDate(dateString)} ${formatTime(dateString, { hour: "numeric", minute: "numeric", second: "numeric" })}`;
