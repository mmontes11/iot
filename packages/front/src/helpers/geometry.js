import PropTypes from "prop-types";

export const pointToLatLng = geometry => {
  if (geometry == null || !geometry.coordinates || geometry.type == null || geometry.type !== "Point") {
    return null;
  }
  const [lng, lat] = geometry.coordinates;
  return { lat, lng };
};

export const PropTypeLatLng = PropTypes.shape({
  lat: PropTypes.number.isRequired,
  lng: PropTypes.number.isRequired,
});
