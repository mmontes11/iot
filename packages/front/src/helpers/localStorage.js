const languageKey = "language";
const tokenKey = "token";

export const setLanguage = language => window.localStorage.setItem(languageKey, language);
export const getLanguage = () => window.localStorage.getItem(languageKey);
export const getToken = () => window.localStorage.getItem(tokenKey);
