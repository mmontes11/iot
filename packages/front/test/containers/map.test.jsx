import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Map from "containers/map";
import { pointToLatLng } from "helpers/geometry";
import { initialState, defaultStore, thing } from "../constants/index";

describe("containers/map", () => {
  it("renders a map", () => {
    const wrapper = mount(
      <Provider store={defaultStore}>
        <Map />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders a map with marker", () => {
    const wrapper = mount(
      <Provider store={defaultStore}>
        <Map marker={{ point: pointToLatLng(thing.geometry) }} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders a map with label and marker", () => {
    const wrapper = mount(
      <Provider store={defaultStore}>
        <Map marker={{ label: thing.name, point: pointToLatLng(thing.geometry) }} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders a map with open dialog", () => {
    const state = {
      ...initialState,
      app: {
        isMapDialogOpened: true,
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <Provider store={store}>
        <Map marker={{ label: thing.name, point: pointToLatLng(thing.geometry) }} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("returns null when trying to convert an invalid geometry", () => {
    expect(pointToLatLng(null)).toBeNull();
    expect(pointToLatLng({ foo: "bar" })).toBeNull();
  });
});
