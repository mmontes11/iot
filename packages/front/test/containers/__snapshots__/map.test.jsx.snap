// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/map renders a map 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(withProps(withScriptjs(withGoogleMap(Map))))>
    <withProps(withScriptjs(withGoogleMap(Map)))
      isDialogOpened={false}
      toggleDialog={[Function]}
    >
      <withScriptjs(withGoogleMap(Map))
        containerElement={
          <div
            className="map"
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
        isDialogOpened={false}
        loadingElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        toggleDialog={[Function]}
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </withScriptjs(withGoogleMap(Map))>
    </withProps(withScriptjs(withGoogleMap(Map)))>
  </Connect(withProps(withScriptjs(withGoogleMap(Map))))>
</Provider>
`;

exports[`containers/map renders a map with label and marker 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(withProps(withScriptjs(withGoogleMap(Map))))
    marker={
      Object {
        "label": "foo",
        "point": Object {
          "lat": 40.21339,
          "lng": -40.58457,
        },
      }
    }
  >
    <withProps(withScriptjs(withGoogleMap(Map)))
      isDialogOpened={false}
      marker={
        Object {
          "label": "foo",
          "point": Object {
            "lat": 40.21339,
            "lng": -40.58457,
          },
        }
      }
      toggleDialog={[Function]}
    >
      <withScriptjs(withGoogleMap(Map))
        center={
          Object {
            "lat": 40.21339,
            "lng": -40.58457,
          }
        }
        containerElement={
          <div
            className="map"
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
        isDialogOpened={false}
        label="foo"
        loadingElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        marker={
          Object {
            "label": "foo",
            "point": Object {
              "lat": 40.21339,
              "lng": -40.58457,
            },
          }
        }
        point={
          Object {
            "lat": 40.21339,
            "lng": -40.58457,
          }
        }
        toggleDialog={[Function]}
        zoom={12}
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </withScriptjs(withGoogleMap(Map))>
    </withProps(withScriptjs(withGoogleMap(Map)))>
  </Connect(withProps(withScriptjs(withGoogleMap(Map))))>
</Provider>
`;

exports[`containers/map renders a map with marker 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(withProps(withScriptjs(withGoogleMap(Map))))
    marker={
      Object {
        "point": Object {
          "lat": 40.21339,
          "lng": -40.58457,
        },
      }
    }
  >
    <withProps(withScriptjs(withGoogleMap(Map)))
      isDialogOpened={false}
      marker={
        Object {
          "point": Object {
            "lat": 40.21339,
            "lng": -40.58457,
          },
        }
      }
      toggleDialog={[Function]}
    >
      <withScriptjs(withGoogleMap(Map))
        center={
          Object {
            "lat": 40.21339,
            "lng": -40.58457,
          }
        }
        containerElement={
          <div
            className="map"
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
        isDialogOpened={false}
        loadingElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        marker={
          Object {
            "point": Object {
              "lat": 40.21339,
              "lng": -40.58457,
            },
          }
        }
        point={
          Object {
            "lat": 40.21339,
            "lng": -40.58457,
          }
        }
        toggleDialog={[Function]}
        zoom={12}
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </withScriptjs(withGoogleMap(Map))>
    </withProps(withScriptjs(withGoogleMap(Map)))>
  </Connect(withProps(withScriptjs(withGoogleMap(Map))))>
</Provider>
`;

exports[`containers/map renders a map with open dialog 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(withProps(withScriptjs(withGoogleMap(Map))))
    marker={
      Object {
        "label": "foo",
        "point": Object {
          "lat": 40.21339,
          "lng": -40.58457,
        },
      }
    }
  >
    <withProps(withScriptjs(withGoogleMap(Map)))
      isDialogOpened={true}
      marker={
        Object {
          "label": "foo",
          "point": Object {
            "lat": 40.21339,
            "lng": -40.58457,
          },
        }
      }
      toggleDialog={[Function]}
    >
      <withScriptjs(withGoogleMap(Map))
        center={
          Object {
            "lat": 40.21339,
            "lng": -40.58457,
          }
        }
        containerElement={
          <div
            className="map"
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
        isDialogOpened={true}
        label="foo"
        loadingElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        mapElement={
          <div
            style={
              Object {
                "height": "100%",
              }
            }
          />
        }
        marker={
          Object {
            "label": "foo",
            "point": Object {
              "lat": 40.21339,
              "lng": -40.58457,
            },
          }
        }
        point={
          Object {
            "lat": 40.21339,
            "lng": -40.58457,
          }
        }
        toggleDialog={[Function]}
        zoom={12}
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </withScriptjs(withGoogleMap(Map))>
    </withProps(withScriptjs(withGoogleMap(Map)))>
  </Connect(withProps(withScriptjs(withGoogleMap(Map))))>
</Provider>
`;
